spec/furniture/marketplace/cart_products_controller_request_spec.rb

```diff
@@ -5,36 +5,117 @@
   let(:space) { marketplace.space }
   let(:room) { marketplace.room }
   let(:member) { create(:membership, space: space).member }
+  let(:document_root_element) { Nokogiri::HTML::Document.parse(response.body) }
+  let(:request_as) { :html }
 
   describe "#create" do
+    subject(:perform_request) do
+      post path, as: request_as, params: {cart_product: {product_id: product.id, quantity: 1}}
+      response
+    end
+
+    let(:product) { create(:marketplace_product, marketplace: marketplace) }
+    let(:cart) { create(:marketplace_cart, marketplace: marketplace) }
+    let(:path) { polymorphic_path(cart.location(child: :cart_products)) }
+
     it "Add a Product to the Cart" do
-      product = create(:marketplace_product, marketplace: marketplace)
-      cart = create(:marketplace_cart, marketplace: marketplace)
+      perform_request
 
-      post polymorphic_path(cart.location(child: :cart_products), params: {cart_product: {product_id: product.id}})
       expect(response).to redirect_to([space, room])
     end
+
+    context "when a turbo stream" do
+      let(:request_as) { :turbo_stream }
+
+      it "Replaces the cart product, cart footer and cart total" do
+        perform_request && cart.reload
+
+        assert_select("turbo-stream[action='replace'][target='cart-product-#{product.id}']") do
+          assert_select("*[data-cart-product-quantity]", text: "1")
+        end
+
+        assert_select("turbo-stream[action='replace'][target='cart-footer-#{cart.id}']")
```

Comment:
I think using the `have_rendered_turbo_stream` matcher here would be fine. One idea to make it more expressive may be to use kwargs for the matcher, so it is clear that you are specifying a `target` and an `action` and whatnot to match against.